--- hivex/ocaml/Makefile.in.orig	2011-05-17 17:38:17.300761216 +0100
+++ hivex/ocaml/Makefile.in	2011-05-17 17:38:27.668965851 +0100
@@ -54,7 +54,8 @@
 @HAVE_OCAMLOPT_TRUE@@HAVE_OCAML_TRUE@am__append_1 = mlhivex.cmxa
 DIST_COMMON = $(srcdir)/.depend $(srcdir)/META.in \
 	$(srcdir)/Makefile.am $(srcdir)/Makefile.in
-@HAVE_OCAML_TRUE@am__append_2 = $(noinst_DATA)
+@HAVE_OCAMLOPT_TRUE@@HAVE_OCAML_TRUE@am__append_2 = *.cmx *.cmxa
+@HAVE_OCAML_TRUE@am__append_3 = $(noinst_DATA)
 subdir = ocaml
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/00gnulib.m4 \
@@ -870,7 +871,7 @@
 	t/*.ml
 
 CLEANFILES = *.cmi *.cmo *.cmx *.cma *.cmxa *.o *.a *.so t/*.cmi \
-	t/*.cmo t/*.cmx t/*.o t/*.a t/*.so $(am__append_2)
+	t/*.cmo t/*.cmx t/*.o t/*.a t/*.so $(am__append_3)
 AM_CPPFLAGS = \
   -I$(top_builddir) -I$(OCAMLLIB) -I$(top_srcdir)/ocaml \
   -I$(top_srcdir)/lib \
@@ -894,6 +895,10 @@
 @HAVE_OCAML_TRUE@	t/hivex_300_fold
 
 @HAVE_OCAML_TRUE@SUFFIXES = .cmo .cmi .cmx .ml .mli .mll .mly
+
+# Do the installation by hand, because we want to run ocamlfind.
+@HAVE_OCAML_TRUE@install_files = META *.so *.a *.cma *.cmi *.mli \
+@HAVE_OCAML_TRUE@	$(am__append_2)
 all: all-am
 
 .SUFFIXES:
@@ -1211,17 +1216,13 @@
 @HAVE_OCAML_TRUE@hivex.cmo: hivex.cmi
 @HAVE_OCAML_TRUE@hivex.cmx: hivex.cmi
 
-# Do the installation by hand, because we want to run ocamlfind.
 @HAVE_OCAML_TRUE@install-data-hook:
 @HAVE_OCAML_TRUE@	mkdir -p $(DESTDIR)$(OCAMLLIB)
 @HAVE_OCAML_TRUE@	mkdir -p $(DESTDIR)$(OCAMLLIB)/stublibs
 @HAVE_OCAML_TRUE@	$(OCAMLFIND) install \
 @HAVE_OCAML_TRUE@	  -ldconf ignore -destdir $(DESTDIR)$(OCAMLLIB) \
-@HAVE_OCAML_TRUE@	  hivex \
-@HAVE_OCAML_TRUE@	  META *.so *.a *.cma *.cmi *.mli
-@HAVE_OCAMLOPT_TRUE@@HAVE_OCAML_TRUE@	$(OCAMLFIND) install \
-@HAVE_OCAMLOPT_TRUE@@HAVE_OCAML_TRUE@	  -ldconf ignore -destdir $(DESTDIR)$(OCAMLLIB) \
-@HAVE_OCAMLOPT_TRUE@@HAVE_OCAML_TRUE@	  *.cmx *.cmxa
+@HAVE_OCAML_TRUE@	  $(PACKAGE_NAME) \
+@HAVE_OCAML_TRUE@	  $(install_files)
 
 # Tell version 3.79 and up of GNU make to not build goals in this
 # directory in parallel.  (See RHBZ#502309).
